hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -687,7 +687,7 @@ <T> T invokeAtAvailableNs(RemoteMethod method, Class<T> clazz)
     // If default Ns is present return result from that namespace.
     if (!nsId.isEmpty()) {
       try {
-        return rpcClient.invokeSingle(nsId, method, clazz);
+        return rpcClient.invokeSingle(nsId, method, clazz, "");
```

Comment:
Can we add an actual test?

Reply:
> > Yes，the src is not the dst path, but we can use the src to lookup real dst path in mount table. Here, we can not execute this.subclusterResolver.getMountPoints(path);, because the statement is time-consuming.
> 
> From correctness point of view it isn't correct. If `/mount/path` points to Ns0 `/dir` and we log` /mount/path` in the namenode, it can lead to confusions if the namenode also has a path named `/mount/path`, If you operate on `/dir` via router `/mount/path` and if you directly operate on `/mount/path` on the namenode.
> 
> If it has serious performance penalties we need to make it configurable and then figure out what can be done when the config is disabled.

Hi, @ayushtkn , can we prepend some specific prefix, such as "RBF#src_path:" to the dir?  In this way, we can distinguish RBF scene from direct namenode scene. What’s your opinion?  I am looking forward to your reply and do the next steps. thanks again~

Reply:
> Can we add an actual test?

OK~ @goiri , I will add actual test soon~ thanks a lot.

Reply:
@goiri will that be ok to have a prefix then path wrt router rather than the actual path wrt namenode

Reply:
Can't we make it an actual field or attribute?

Reply:
Hi, @goiri , here if we make it an actual field or attribute, It would invoke `getLocationsForPath` to get actual path.
Then,  It violates the purpose that we use previous block to improve some RPC performance.

Reply:
Hi，@goiri ,  could you please also review the code of https://issues.apache.org/jira/browse/HDFS-16865

Reply:
Hi, @ayushtkn @goiri . Could you please take a look at https://issues.apache.org/jira/browse/HDFS-16882
is it feasible？ thanks a lot.

Reply:
I meant to make it part of RemoteMethod.